modernforms/modernforms/modernforms_fan.py

```diff
@@ -72,6 +99,17 @@ def fan_on(self, state: bool) -> None:
         """
         self.set_device_state({"fanOn": state})
 
+    def toggleFan(self, direction: bool = seasonDirection):
+        """ Toggles the fan state. 
+        
+        Also automatically detects which season it is and spins the fan in the corresponding direction for that season.
+        This can be overriden by setting the direction paramter to something else.
+        """
+        
+        state = not self._data['fanOn']
+        self.set_device_state({"fanOn": state})
+        self.set_device_state({"fanDirection": direction})
```

Comment:
Please combine into one call to set_device_state. The API on the fans isn't very robust and 2 quick success calls can be unsuccessful, especially on the revision 2 of the fan hardware.

Reply:
Got it, I combined that into one call.